invenio_circulation/errors.py

```diff
@@ -8,50 +8,142 @@
 
 """Circulation exceptions."""
 
+import json
 
-class CirculationException(Exception):
-    """Exceptions raised by circulation module."""
+from flask import current_app
+from invenio_rest.errors import RESTException
 
-    def __init__(self, msg):
-        """Constructor."""
-        self.msg = msg
 
+class CirculationException(RESTException):
+    """Base Exception for circulation module, inherit, don't raise."""
 
-class InvalidState(CirculationException):
-    """State not found in circulation configuration."""
+    code = 400
 
+    @property
+    def name(self):
+        """The status name."""
+        return type(self).__name__
 
-class ItemNotAvailable(CirculationException):
-    """Exception raised from action on unavailable item."""
+    def __init__(self, **kwargs):
+        """Initialize exception."""
+        super(CirculationException, self).__init__(**kwargs)
 
+    def get_response(self, environ=None, **kwargs):
```

Comment:
I think it is better to override `get_body` rather than `get_response`, no?
See it here: https://github.com/inveniosoftware/invenio-rest/blob/master/invenio_rest/errors.py#L68

Example pseudo-code:
```
def get_body(...):
    string_body = super.get_body()
    body = json.loads(string_body)
    body["error_module"] = "Circulation"
    body["error_class"] = self.name
    return json.dumps(body)
```